### PR TITLE
Update release process doc for GH actions

### LINKED ISSUE #1831: Update release process guide
### What problem are you facing?

The Crossplane [release process guide](https://crossplane.io/docs/master/release/release-process) is out of date and makes references to operations such as releasing stacks, which are no longer relevant.

### How could Crossplane help solve your problem?

The release process should be updated to the extent that someone who is reasonably familiar with the Crossplane ecosystem can run a release.

### PR DESCRIPTION
**Pull request by hasheddan**
### Description of your changes

Updates release process doc to reflect current operations and usage of Github Actions.

Fixes #1831 

Note: Needs backport to `release-0.14`

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

n/a

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: docs/release/release-process.md [hunk 1]
```diff
@@ -246,198 +176,42 @@ and collect feedback.  Do **not** hit "Publish release" yet.
...
-`v0.5.0`) and the channel you'd like to promote it to.  The first time you run
-this pipeline on a new release branch, you will not be prompted for values, so
-the pipeline will fail. Just run (not replay) it a second time to be prompted.
+Run the `Promote` action on the release branch and input the version you would
+like to promote (e.g. `v0.5.0`) and the channel you'd like to promote it to.
 
-* Run `promote` pipeline for `master` channel
-* Run `promote` pipeline for `alpha` channel
+* Run `Promote` action for `master` channel on release branch
+* Run `Promote` action for `alpha` channel on release nramch
```
Comment (negz) [inline docs/release/release-process.md:212]:
```suggestion
* Run `Promote` action for `alpha` channel on release branch
```

**Review by negz — APPROVED** [review]
LGTM! It seems like this no longer discusses releasing providers. Are you planning to document that elsewhere, or do you feel like this is intuitive enough?

File: docs/release/release-process.md (hunk 1 above)
Comment (hasheddan) [reply docs/release/release-process.md:212]: Ah thanks for catching this!

**Comment by hasheddan** [comment]
@negz the idea here is that all repos now run the same release process. I think providers pretty much look the same as core Crossplane based on the process we followed for v0.14.

**Pull request merged**